pkg/kubelet/server.go

```diff
@@ -699,24 +769,24 @@ type StatsRequest struct {
 }
 
 // serveStats implements stats logic.
-func (s *Server) serveStats(w http.ResponseWriter, req *http.Request) {
+func (s *Server) serveStats(request *restful.Request, response *restful.Response) {
 	// Stats requests are in the following forms:
 	//
 	// /stats/                                              : Root container stats
 	// /stats/container/                                    : Non-Kubernetes container stats (returns a map)
 	// /stats/<pod name>/<container name>                   : Stats for Kubernetes pod/container
 	// /stats/<namespace>/<pod name>/<uid>/<container name> : Stats for Kubernetes namespace/pod/uid/container
-	components := strings.Split(strings.TrimPrefix(path.Clean(req.URL.Path), "/"), "/")
+	components := strings.Split(strings.TrimPrefix(path.Clean(request.Request.URL.Path), "/"), "/")
 	var stats interface{}
 	var err error
-	var query StatsRequest
-	query.NumStats = 60
-
-	err = json.NewDecoder(req.Body).Decode(&query)
-	if err != nil && err != io.EOF {
-		s.error(w, err)
+	query := new(StatsRequest)
+	if err := request.ReadEntity(query); err != nil {
+		response.WriteError(http.StatusBadRequest, err)
 		return
 	}
+
+	query.NumStats = 60
```

Comment:
Previously, NumStats was set to be the initial limit before updating it based on the user request. Now IIUC, you are overriding user request. So essentially we should set it only if the NumStats is not set by the user.